hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/ContainerKeyMapperHelper.java

```diff
@@ -84,66 +124,108 @@ public static boolean reprocess(OMMetadataManager omMetadataManager,
+              Thread.currentThread().getId(), k -> new ConcurrentHashMap<>());
+          
+          handleKeyReprocess(kv.getKey(), kv.getValue(), myLocalMap, SHARED_CONTAINER_KEY_COUNT_MAP,
               reconContainerMetadataManager);
-          omKeyCount++;
 
-          // Check and flush data if it reaches the batch threshold
-          if (!checkAndCallFlushToDB(containerKeyMap, containerKeyFlushToDBMaxThreshold,
+          // Flush this worker's map when it reaches threshold
+          if (myLocalMap.size() >= PER_WORKER_THRESHOLD) {
+            synchronized (flushLock) {
+              if (!flushAndCommitContainerKeyInfoToDB(myLocalMap, Collections.emptyMap(),
+                  reconContainerMetadataManager)) {
+                throw new UncheckedIOException(new IOException("Unable to flush containerKey information to the DB"));
+              }
+            }
+          }
+          return null;
+        } catch (IOException e) {
+          throw new UncheckedIOException(e);
+        }
+      };
+      
+      try (ParallelTableIteratorOperation<String, OmKeyInfo> keyIter =
+               new ParallelTableIteratorOperation<>(omMetadataManager, omKeyInfoTable,
+                   StringCodec.get(), maxIterators, maxWorkers, maxKeysInMemory, PER_WORKER_THRESHOLD)) {
+        keyIter.performTaskOnTableVals(taskName, null, null, kvOperation);
+      }
+
+      // Final flush: Write remaining entries from all worker local maps to DB
+      for (Map<ContainerKeyPrefix, Integer> workerLocalMap : allLocalMaps.values()) {
+        if (!workerLocalMap.isEmpty()) {
+          if (!flushAndCommitContainerKeyInfoToDB(workerLocalMap, Collections.emptyMap(),
               reconContainerMetadataManager)) {
-            LOG.error("Failed to flush container key data for {}", taskName);
+            LOG.error("Failed to flush worker local map for {}", taskName);
             return false;
           }
         }
       }
 
-      // Final flush and commit
-      if (!flushAndCommitContainerKeyInfoToDB(containerKeyMap, containerKeyCountMap, reconContainerMetadataManager)) {
-        LOG.error("Failed to flush Container Key data to DB for {}", taskName);
+      // Capture total container count from shared map
+      long totalContainers = SHARED_CONTAINER_KEY_COUNT_MAP.size();
+
+      // Final flush: Shared container count map
+      if (!flushAndCommitContainerKeyInfoToDB(Collections.emptyMap(), SHARED_CONTAINER_KEY_COUNT_MAP, reconContainerMetadataManager)) {
+        LOG.error("Failed to flush shared container count map for {}", taskName);
         return false;
       }
 
+      // Write total container count once at the end (after all processing)
+      if (totalContainers > 0) {
+        reconContainerMetadataManager.incrementContainerCountBy(totalContainers);
+      }
+
+      // Decrement active task counter and cleanup if this is the last task
+      int remainingTasks = ACTIVE_TASK_COUNT.decrementAndGet();
```

Comment:
perfrom this in INITIALIZATION_LOCK, decrement and validate if count is "0", if "0", perform update

Reply:
The ACTIVE_TASK_COUNT.decrementAndGet() operation is already atomic and thread-safe.